src/main/java/nl/tudelft/jpacman/level/Level.java

```diff
@@ -349,9 +385,121 @@ public void run() {
 	}
 
 	/**
-	 * An observer that will be notified when the level is won or lost.
+	 * A task that moves the single player and reschedules itself after it
+	 * finished.
+	 * 
+	 * @author bellafkih
+	 */
+	private final class PacmanMoveTask implements Runnable {
+
+		/**
+		 * The service executing the task.
+		 */
+		private final ScheduledExecutorService service;
+
+		/**
+		 * The player to move.
+		 */
+		private final Player p;
+
+		/**
+		 * Creates a new task.
+		 * 
+		 * @param s
+		 *            The service that executes the task.
+		 * @param p
+		 *            The player to move.
+		 */
+		private PacmanMoveTask(ScheduledExecutorService s, Player p) {
+			this.service = s;
+			this.p = p;
+		}
+
+		@Override
+		public void run() {
+			Direction currentDirection = this.p.getDirection();
+			List<Direction> listOfPossibleDirection = new ArrayList<Direction>();
+			Square leftSquare = board.squareAt(getCurrentPacManPosition()
+					.getX() - 1, getCurrentPacManPosition().getY());
+
+			Square rightSquare = board.squareAt(getCurrentPacManPosition()
+					.getX() + 1, getCurrentPacManPosition().getY());
+
+			Square northSquare = board.squareAt(getCurrentPacManPosition()
+					.getX(), getCurrentPacManPosition().getY() - 1);
+
+			Square southSquare = board.squareAt(getCurrentPacManPosition()
+					.getX(), getCurrentPacManPosition().getY() + 1);
+			switch (currentDirection) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539212/issues/source?bid=3256381&fileBranchId=3281826#l433)